sanic/http.py

```diff
@@ -218,7 +218,7 @@ async def http1_request_header(self):
             raise InvalidUsage("Bad Request")
 
         headers_instance = Header(headers)
-        self.upgrade_websocket = headers_instance.get("upgrade") == "websocket"
+        self.upgrade_websocket = "upgrade" in headers_instance and headers_instance.get("upgrade", None) is not None and headers_instance.get("upgrade").lower() == "websocket"
```

Comment:
If we really want to do this check the easier might be 

```suggestion
        self.upgrade_websocket =  headers_instance.get("upgrade", "").lower() == "websocket"
```

Reply:
I like this better